torch/_export/utils.py

```diff
@@ -821,6 +821,14 @@ def _extract_pytree_key(x):
 
     # map user input names with mod.forward() signature
     combined_args = _bind_signature_to_inputs(mod, fake_args, fake_kwargs)
+    # reorder following args, kwargs
+    combined_args = {
+        name: combined_args[name]
```

Comment:
just curious why the order in the dictionary matters? If the order matters, maybe we should make it an ordered dict to be explicit?

Reply:
Use itertools.chain. No need for this weird list unpacking

Reply:
Turns out that the graph signature has user inputs in the order they were received, but combined args orders them in the order of the original forward signature. The actual ordering doesn't matter so much as that they should be consistent, because we zip them soon afterwards.

Reply:
@yushangdi I ended up changing the implementation a bit because it was not accounting for variadic kwargs. Latest update matches what happens in strict.

Given a module signature: `def forward(a, b, c, kwargs): ...`
and args: `a_val`, kwargs: `{c=c_val, b=b_val, d=d_val}`

we should now get placeholder names `a, c, b, d`. So overall preserving call site order instead of def site order.